semconv/exception.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package semconv
+
+import "go.opentelemetry.io/otel/attribute"
+
+// Semantic conventions for exception attribute keys.
+const (
+	// The type of the exception (its fully-qualified class name, if
+	// applicable). The dynamic type of the exception should be preferred
+	// over the static type in languages that support it.
+	ExceptionTypeKey = attribute.Key("exception.type")
+
+	// The exception message
+	ExceptionMessageKey = attribute.Key("exception.message")
+
+	// A stacktrace as a string in the natural representation for the
+	// language runtime. The representation is to be determined and
+	// documented by each language SIG.
+	ExceptionStacktraceKey = attribute.Key("exception.stacktrace")
+
+	// SHOULD be set to true if the exception event is recorded at a
+	// point where it is known that the exception is escaping the scope
+	// of the span.
```

Comment:
Can we make these comments relevant to this project?

```suggestion
	// The Go type containing the error or exception.
	ExceptionTypeKey = attribute.Key("exception.type")

	// The exception message.
	ExceptionMessageKey = attribute.Key("exception.message")

	// A stacktrace as a string. This most commonly will come from
	// "runtime/debug".Stack.
	ExceptionStacktraceKey = attribute.Key("exception.stacktrace")

	// If the exception event is recorded at a point where it is known
	// that the exception is escaping the scope of the span this
	// attribute is set to true.
```

Reply:
I'll go ahead and accept these changes, but question regarding the `exception.escaped`:

```
// If the exception event is recorded at a point where it is known
// that the exception is escaping the scope of the span this
// attribute is set to true.
```

I don't believe we set this on panic as part of this PR -- is that a change you want to see in this PR?  The language sounds like we'll add it on behalf of the user, rather than something the user should use.

Reply:
> I'll go ahead and accept these changes, but question regarding the `exception.escaped`:
> 
> ```
> // If the exception event is recorded at a point where it is known
> // that the exception is escaping the scope of the span this
> // attribute is set to true.
> ```
> 
> I don't believe we set this on panic as part of this PR -- is that a change you want to see in this PR? The language sounds like we'll add it on behalf of the user, rather than something the user should use.

Please feel free to make further revisions. My suggestion was to describe _what_ that key will be used for, in that _if_ the situation occurs where an exception event is recorded when escaping a scope that attribute will be used to describe it and it will be set to true (if the implementation follow the semantic conventions).

If that's not coming through, please feel free to update.